File: view/android/VerificationCodeJavaDemo/app/src/main/java/com/example/verificationcodejavademo/network/ServerApi.java
package com.example.verificationcodejavademo.network;


import com.example.verificationcodejavademo.model.CaptchaCheckIt;
import com.example.verificationcodejavademo.model.CaptchaGetIt;
import com.example.verificationcodejavademo.model.WordCaptchaGetIt;


import io.reactivex.Observable;
import okhttp3.RequestBody;
import retrofit2.http.Body;
import retrofit2.http.POST;

/**
 * Date:2020/5/18
 * author:wuyan
 */
public interface ServerApi {

    final static String BaseUrl = "https://captcha.anji-plus.com/captcha-api/";
//    final static String BaseUrl = "http://10.108.12.11:8080/";

    //获取验证码
    @POST("captcha/get")
    Observable<BaseResponse<CaptchaGetIt>> getAsync(@Body RequestBody body);

    //获取文字的验证码
    @POST("captcha/get")
    Observable<BaseResponse<WordCaptchaGetIt>> getWordCaptchaAsync(@Body RequestBody body);

    //核对验证码
    @POST("captcha/check")
    Observable<BaseResponse<CaptchaCheckIt>> checkAsync(@Body RequestBody body);
}

File: view/android/VerificationCodeDemo/app/src/main/java/com/example/verificationcodedemo/network/ServerApi.kt
package com.example.verificationcodedemo.network

import com.example.verificationcodedemo.model.*
import kotlinx.coroutines.Deferred
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST

/**
 * Date:2020/4/29
 * author:wuyan
 */
interface ServerApi {

    companion object {
        const val urlDefault: String = "https://captcha.anji-plus.com/captcha-api/"
    }

    //获取验证码
    @POST("captcha/get")
    fun getAsync(@Body body: CaptchaGetOt): Deferred<Response<Input<CaptchaGetIt>>>

    //获取文字的验证码
    @POST("captcha/get")
    fun getWordCaptchaAsync(@Body body: CaptchaGetOt): Deferred<Response<Input<WordCaptchaGetIt>>>

    //核对验证码
    @POST("captcha/check")
    fun checkAsync(@Body body: CaptchaCheckOt): Deferred<Response<Input<CaptchaCheckIt>>>

}